src/SFML/Window/GlContext.cpp

```diff
@@ -399,10 +400,22 @@ void GlContext::registerContextDestroyCallback(ContextDestroyCallback callback,
 ////////////////////////////////////////////////////////////
 void GlContext::acquireTransientContext()
 {
+    using GlContextImpl::currentContext;
+    using GlContextImpl::currentContextTransientCount;
     using GlContextImpl::mutex;
     using GlContextImpl::TransientContext;
     using GlContextImpl::transientContext;
 
+    // Fast path if we already have a context active on this thread
+    if (currentContext)
+    {
+        ++currentContextTransientCount;
```

Comment:
It feels like we're approximating a shared pointer. Does `currentContextTransientCount` not constitute a ref count?

Reply:
Both `currentContext` and `currentContextTransientCount` are `thread_local` so it lets us skip the expensive locking that would otherwise be required. Also `currentContext` is non-owning. `currentContextTransientCount` just helps us keep track of how many times we are "borrowing" the `currentContext`, there are no ownership semantics in play.